src/util/rocksdb/mod.rs

```diff
@@ -335,6 +347,61 @@ pub fn compact_range(
     db.compact_range_cf_opt(handle, &compact_opts, start_key, end_key);
 }
 
+pub fn compact_files_in_range(db: &DB, start_key: &[u8], end_key: &[u8]) -> Result<(), String> {
+    for cf_name in db.cf_names() {
+        compact_files_in_range_cf(db, cf_name, start_key, end_key)?;
+    }
+    Ok(())
+}
+
+pub fn compact_files_in_range_cf(
+    db: &DB,
+    cf_name: &str,
+    start_key: &[u8],
+    end_key: &[u8],
+) -> Result<(), String> {
+    let cf = db.cf_handle(cf_name).unwrap();
+    let cf_opts = db.get_options_cf(cf);
+    let output_level = cf_opts.get_num_levels() - 1;
+    let output_compression = cf_opts
+        .get_compression_per_level()
+        .get(output_level)
+        .cloned()
+        .unwrap_or(DBCompressionType::No);
+    let output_file_size_limit = cf_opts.get_target_file_size_base();
+
+    let mut input_files = Vec::new();
+    let mut all_files_in_output_level = true;
+    let cf_meta = db.get_column_family_meta_data(cf);
```

Comment:
What if some rocksdb compaction happens during the loop and some files are deleted or added, will the metadata still be correct?

Reply:
It will not be correct and the compaction will fail. It only means to be used in import mode, when compaction is disabled.

Reply:
So we need assert that compaction is disabled before execute following codes?

Reply:
I don't think we should assert here. If it can't compact it will just fail and the client should handle that.